src/components/TodoList/TodoList.tsx

```diff
@@ -1,12 +1,61 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import classNames from 'classnames';
+import { useAppSelector } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+import { actions as currentTodoActions } from '../../features/currentTodo';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useDispatch();
+  const {
+    todos,
+    filter: {
+      status,
+      query,
+    },
+    currentTodo,
+  } = useAppSelector((state) => state);
+
+  const handleEye = (todo: Todo) => {
+    dispatch(currentTodoActions.setTodo(todo));
+  };
+
+  const filteredTodos = useMemo(() => {
+    let copyTodo = [...todos];
+
+    switch (status) {
+      case 'active':
+        copyTodo = copyTodo.filter((todo) => !todo.completed);
+        break;
+
+      case 'completed':
+        copyTodo = copyTodo.filter((todo) => todo.completed);
+        break;
+
+      case 'all':
+        break;
+
+      default:
+        return todos;
+    }
+
+    if (query) {
+      copyTodo = copyTodo.filter((todo) => {
+        return todo.title.toLowerCase().includes(query.toLowerCase().trim());
+      });
+    }
+
+    return copyTodo;
+  }, [todos, status, query]);
```

Comment:
```suggestion
const filteredTodos = useMemo(() => {
  const isMatch = (title) => query ? title.toLowerCase().includes(query.toLowerCase().trim()) : true;

  return todos.filter(({ title, completed }) => {
    switch (status) {
      case 'active':
        return !completed && isMatch(title);

      case 'completed':
        return completed && isMatch(title);

      default:
        return isMatch(title);
    }
  });
}, [todos, status, query]);
```


please have a look at this implementation. your option is not a mistake though. But with this one we write less code and duplicates